File: 프로그래머스/lv2/이재표/프로세스.java
import java.util.*;
class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        Queue<int[]>queue=new LinkedList<>();
        for(int i=0;i<priorities.length;i++){
            queue.add(new int[]{priorities[i],i});
        }
        while(!queue.isEmpty()){
            int[] current=queue.poll();
            boolean isFlag=false;
            int cnt=0;
            for(int[]q:queue){
                cnt++;
                if(q[0]>current[0]){
                    isFlag=true;
                }
            }
            if(isFlag){
                queue.offer(current);
            }else{
                if(current[1]==location){
                    return priorities.length-cnt;
                }
            }
        }
        return answer;
    }
}

File: 프로그래머스/lv2/이하은/프로세스.kt
import java.util.*
typealias PII = Pair<Int, Int>
    
class Solution {
    fun solution(priorities: IntArray, location: Int): Int {
        val q = ArrayDeque<PII>()
        for((i, priority) in priorities.withIndex()){
            q.offer(Pair(i, priority))
        }
        
        var cnt = 0
        while(!q.isEmpty()){
            val now = q.poll()
            
            // 뒤쪽에 우선순위가 더 높은 게 있다면 
            if(q.filter { now.second < it.second }.size > 0){
                // 큐에서 꺼낸 걸 뒤에 넣는다. 
                q.offer(now)
            }else{
                cnt++ // 실행한 프로세스 개수 갱신 
                if(now.first == location) break 
            }
        }
        
        return cnt 
    }
}